src/python_bindings/bindings.cpp

```diff
@@ -127,13 +127,34 @@ class PyHNSWLibIndex : public PyVecSimIndex {
     PyHNSWLibIndex(const std::string &location, const HNSWParams *hnsw_params = nullptr) {
         this->index = HNSWFactory::NewIndex(location, hnsw_params);
     }
-
     void setDefaultEf(size_t ef) {
-        auto *hnsw = reinterpret_cast<HNSWIndex<float, float> *>(index);
-        hnsw->setEf(ef);
+        VecSimType data_type = this->index->info().hnswInfo.type;
+        if (data_type == VecSimType_FLOAT32) {
+            setDefaultEfImp<float>(ef);
+        } else if (data_type == VecSimType_FLOAT64) {
+            setDefaultEfImp<double>(ef);
+        }
     }
+
     void saveIndex(const std::string &location) {
-        auto *hnsw = reinterpret_cast<HNSWIndex<float, float> *>(index);
+        VecSimType data_type = this->index->info().hnswInfo.type;
+        if (data_type == VecSimType_FLOAT32) {
+            saveIndexImp<float>(location);
+        } else if (data_type == VecSimType_FLOAT64) {
+            saveIndexImp<double>(location);
```

Comment:
Now that you added the capability to save FP64 index via our bindings, you should probably test it in the flow test...